Shopify/cli
# Update dependencies to ESM and change module system to `NodeNext` in `tsconfig.json`

### WHY are these changes introduced?
Migrating external dependencies to their ESM allows loading them concurrently with the rest of the modules of the CLI graph. This PR migrates some of them and addresses some issues that arouse after changing the module system to `NodeNext` in the `tsconfig.json`. This was necessary for Typescript to resolve the imports through the Node convention of declaring exports in the `exports` attribute of the `package.json`.

### WHAT is this pull request doing?

Use the ESM version of the following dependencies:

- [ ] Bugsnag
- [x] color-json
- [x] conf
- [x] envfile
   - I opened [a PR](https://github.com/bevry/envfile/pull/210) to add a missing `exports` key.
- [x] get-port-please
- [ ] simple-git
  - Something is wrong with the generated types because we get the following error [(similar issue)](https://github.com/steveukx/git-js/issues/804):

```
Type 'typeof import("/Users/pepicrft/src/github.com/Shopify/cli/node_modules/.pnpm/simple-git@3.16.0/node_modules/simple-git/dist/typings/index")' has no call signatures.
```

### How to test your changes?
Try to run app and theme commands. If changes in how we import modules lead to runtime errors those should arise when loading the module graph as part of the oclif manifest generation.


### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
